weblab/repocache/entities.py

```diff
@@ -0,0 +1,39 @@
+from django.db.models import F, Max
+
+from .models import CachedEntityVersion
+
+
+def get_public_entity_ids():
+    """
+    Get IDs of all publicly visible entities
+
+    :return: set of entity IDs
+    """
+    return set(
+        CachedEntityVersion.objects.annotate(
+            latest_ts=Max('entity__versions__timestamp')
+        ).filter(
+            timestamp=F('latest_ts'),
+            visibility='public',
```

Comment:
How will this behave with an entity where (say) the latest version is private but the previous version is public?

Ideally you should still return this entity, but when other code displays a version it then they only show the public version, not the latest version. However, we could leave implementing that for a second PR, since it's probably non-trivial and not that common a use case.

Reply:
Yeah at the moment it won't include that entity at all (there's a test case for this). I'll leave it for now but that sounds like a sensible feature to add at some point.